qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -420,6 +420,7 @@ def _build_vqd_result() -> VQDResult:
 
     @staticmethod
     def _update_vqd_result(result, opt_result, eval_time, ansatz) -> VQDResult:
+        # TODO: are result fields lists or ndarrays (or both?)
```

Comment:
As I already mentioned, the result field types are not really respected during population of the fileds

Reply:
Yeah it seems a bit inconsistent, as in `_build_vqd_result`, lists are used but then they are populated with arrays. Personally I think lists would make more sense but since the `EigensolverResult` is using arrays, and the code is functionally also using arrays, I think it would be better to keep supporting them.

Reply:
I'm not sure if there is a concrete suggestion for fix (or leave it as is for now)?

Reply:
I think if we could consistently use `np.array` that would be good, since that's what the original `EigensolverResult` class is using 🙂